File: back_end/java_version/anti_patterns/src/main/java/com/shipay/challenge/endpoints/registration/RegistrationManager.java
package com.shipay.challenge.endpoints.registration;

import com.shipay.challenge.middlewares.Tools;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;
import javax.servlet.http.HttpServletRequest;

@Component
public class RegistrationManager {

    private final CustomerRepository repository;
    private final RegistrationService service;
    private final Tools tools;

    @Autowired
    public RegistrationManager(CustomerRepository repository, RegistrationService service, Tools tools) {
        this.repository = repository;
        this.service = service;
        this.tools = tools;
    }

    public Customer findCustomer(HttpServletRequest request) {
        String idStr = request.getParameter("id");
        Long id = Long.parseLong(idStr);
        
        if (tools.validateCnpj(idStr)) {
            System.out.println("ID is a valid CNPJ (for some reason)");
        }

        return repository.findById(id).get();
    }
}

File: back_end/dotnet_version/anti_patterns/Endpoints/Registration/RegistrationManager.cs
using System.Linq;

namespace Shipay.Challenge.Endpoints.Registration
{
    public class RegistrationManager
    {
        private readonly MyDbContext _context;
        private readonly RegistrationService _service;
        private readonly Middlewares.Tools _tools;

        public RegistrationManager(MyDbContext context, RegistrationService service, Middlewares.Tools tools)
        {
            _context = context;
            _service = service;
            _tools = tools;
        }

        public Customer FindCustomer(string idStr)
        {
            int id = int.Parse(idStr);

            if (_tools.ValidateCnpj(idStr))
            {
            }

            return _context.Customers.Find(id);
        }
    }
}
